### ISSUE TITLE
DMSF install to Redmine 3.0.3   problem

### ISSUE THREAD

**Issue by AlexiaChen** [danmunn/redmine_dmsf#394]
I just download DMSF version 1.5.1.

then, I just move the DMSF folder to the redmine plugin folder and run "bundle install" .the Bitnami Redmine Stack enviroment console show the message as follows:

Warning: this Gemfile contains multiple primary sources. Using `source` more tha
n once without a block is a security risk, and may result in installing unexpect
ed gems. To resolve this warning, use a block to indicate which gems should come
 from the secondary source. To upgrade this warning to an error, run `bundle con
fig disable_multisource true`.
You are trying to install in deployment mode after changing
your Gemfile. Run `bundle install` elsewhere and add the
updated Gemfile.lock to version control.

If this is a development machine, remove the Gemfile freeze
by running `bundle install --no-deployment`.

You have added to the Gemfile:
* rubyzip (>= 1.0.0)
* zip-zip
* simple_enum
* uuidtools (~> 2.1.1)
* dav4rack (~> 0.3.0)
* nokogiri (>= 1.5.10)
* xapian-full-alaveteli

So, I run "bundle show" this command ,console shows : 
Gems included by the bundle:
Could not find rubyzip-1.1.7 in any of the sources

and I run "rake redmine:plugins:migrate RAILS_ENV=production" this command ,it also shows up:
Could not find rubyzip-1.1.7 in any of the sources
Run `bundle install` to install missing gems.

according to the hint above, run "bundle install"   doesn't work fine.  

Does anyone get resolution to solve this problem??   I just need to install DMSF plugin into Redmine

**Comment by u2mike** [comment]
hello all

i have the same/similar Problem ... i did an bitnami stack upgrade to redmine version 3.0.3.stable
and then i wanted to install the dmsf plugin ( i didn´t have this plugin befor upgrading!)
after unzip and running the installation in console with the following command:
    bundle install --without development test postgresql sqlite --no-deployment

i got the following error:

An error occurred while installing xapian-full-alaveteli (1.2.9.7), and Bundler cannot continue.
Make sure that `gem install xapian-full-alaveteli -v '1.2.9.7'` succeeds before bundling.

is there a chance to get this problem solved in near time?

thx!

**Comment by picman** [comment]
Yes, leave Bitnami and use Linux on your server instead.
The reported problem is not related to DMSF but to xapian-full-alivateli.

**Comment by u2mike** [comment]
okay ... first thx for your quick response 
... but let me say that i could finish the installation after installing the following package: 
apt-get install uuid-dev

so i was able to activate the plugin:
bundle install --without development test postgresql sqlite --no-deployment
bundle install --without development test postgresql sqlite --deployment
ruby bin/rake redmine:plugins RAILS_ENV=production

AND: i also could (re)start redmine ... 
but when i click on the every button e.g projekt "settings" button or "administration" button
i get the following (debug)message:

maybe there is a small change to give me a solution for my (or all other bitnami stack) users :-)

### cut start message ###
Started GET "/redmine/plugin_assets/redmine_dmsf/stylesheets/dmsf.css" for 10.105.100.101 at 2015-06-23 05:37:37 +0000

ActionController::RoutingError (No route matches [GET] "/plugin_assets/redmine_dmsf/stylesheets/dmsf.css"):
  vendor/bundle/ruby/2.0.0/gems/actionpack-4.2.1/lib/action_dispatch/middleware/debug_exceptions.rb:21:in `call'
  vendor/bundle/ruby/2.0.0/gems/actionpack-4.2.1/lib/action_dispatch/middleware/show_exceptions.rb:30:in `call'
  vendor/bundle/ruby/2.0.0/gems/railties-4.2.1/lib/rails/rack/logger.rb:38:in `call_app'
  vendor/bundle/ruby/2.0.0/gems/railties-4.2.1/lib/rails/rack/logger.rb:20:in `block in call'
  vendor/bundle/ruby/2.0.0/gems/activesupport-4.2.1/lib/active_support/tagged_logging.rb:68:in `block in tagged'
  vendor/bundle/ruby/2.0.0/gems/activesupport-4.2.1/lib/active_support/tagged_logging.rb:26:in `tagged'
  vendor/bundle/ruby/2.0.0/gems/activesupport-4.2.1/lib/active_support/tagged_logging.rb:68:in `tagged'
  vendor/bundle/ruby/2.0.0/gems/railties-4.2.1/lib/rails/rack/logger.rb:20:in `call'
  vendor/bundle/ruby/2.0.0/gems/actionpack-4.2.1/lib/action_dispatch/middleware/request_id.rb:21:in `call'
  vendor/bundle/ruby/2.0.0/gems/rack-1.6.0/lib/rack/methodoverride.rb:22:in `call'
  vendor/bundle/ruby/2.0.0/gems/rack-1.6.0/lib/rack/runtime.rb:18:in `call'
  vendor/bundle/ruby/2.0.0/gems/activesupport-4.2.1/lib/active_support/cache/strategy/local_cache_middleware.rb:28:in `call'
  vendor/bundle/ruby/2.0.0/gems/actionpack-4.2.1/lib/action_dispatch/middleware/static.rb:113:in `call'
  vendor/bundle/ruby/2.0.0/gems/rack-1.6.0/lib/rack/sendfile.rb:113:in `call'
  vendor/bundle/ruby/2.0.0/gems/railties-4.2.1/lib/rails/engine.rb:518:in `call'
  vendor/bundle/ruby/2.0.0/gems/railties-4.2.1/lib/rails/application.rb:164:in `call'
  vendor/bundle/ruby/2.0.0/gems/railties-4.2.1/lib/rails/railtie.rb:194:in `public_send'
  vendor/bundle/ruby/2.0.0/gems/railties-4.2.1/lib/rails/railtie.rb:194:in `method_missing'
  vendor/bundle/ruby/2.0.0/gems/rack-1.6.0/lib/rack/urlmap.rb:66:in `block in call'
  vendor/bundle/ruby/2.0.0/gems/rack-1.6.0/lib/rack/urlmap.rb:50:in `each'
  vendor/bundle/ruby/2.0.0/gems/rack-1.6.0/lib/rack/urlmap.rb:50:in `call'
  passenger (5.0.6) lib/phusion_passenger/rack/thread_handler_extension.rb:85:in `process_request'
  passenger (5.0.6) lib/phusion_passenger/request_handler/thread_handler.rb:149:in `accept_and_process_next_request'
  passenger (5.0.6) lib/phusion_passenger/request_handler/thread_handler.rb:110:in `main_loop'
  passenger (5.0.6) lib/phusion_passenger/request_handler.rb:414:in `block (3 levels) in start_threads'
  passenger (5.0.6) lib/phusion_passenger/utils.rb:111:in `block in create_thread_and_abort_on_exception'

### cut end message ###

**Comment by picman** [comment]
Okay. Let's leave it open. Somebody might help.

**Comment by u2mike** [comment]
update:
now it works! 

what i have done!

first i have to excuse my rookie doing, because i installed the redmine_dmsf-master.zip version!!!
after i fetched the right version (with git clone ...) and do installing ... i got my verions worked.
additional information --> i had to install the uuid-dev package manually on my bitname stack  (apt-get upgrade uuid-dev)

but ... not ALL works well .... i have the following error by searching in the dms database as follow:
when i want to search in the DMS DB i got the following error:
Internal error

## error start ##
An error occurred on the page you were trying to access.
If you continue to experience problems please contact your Redmine administrator for assistance.
If you are the Redmine administrator, check your log files for details about the error.
## error end ##

this is the output of my debugging log:
## start debug-log##
SELECT COUNT(*) FROM `dmsf_files` WHERE (projects.status <> 9 AND projects.id IN (SELECT em.project_id FROM enabled_modules em WHERE em.name='dmsf') AND dmsf_files.project_id IN (34) AND dmsf_files.deleted = 0) AND (((LOWER(dmsf_files.name) LIKE '%dmsf%') OR (LOWER(dmsf_file_revisions.title) LIKE '%dmsf%') OR (LOWER(dmsf_file_revisions.description) LIKE '%dmsf%')))^[[0m
Mysql2::Error: Unknown column 'projects.status' in 'where clause': SELECT COUNT(*) FROM `dmsf_files` WHERE (projects.status <> 9 AND projects.id IN (SELECT em.project_id FROM enabled_modules em WHERE em.name='dmsf') AND dmsf_files.project_id IN (34) AND dmsf_files.deleted = 0) AND (((LOWER(dmsf_files.name) LIKE '%dmsf%') OR (LOWER(dmsf_file_revisions.title) LIKE '%dmsf%') OR (LOWER(dmsf_file_revisions.description) LIKE '%dmsf%')))
Completed 500 Internal Server Error in 8ms (ActiveRecord: 1.7ms)

ActiveRecord::StatementInvalid (Mysql2::Error: Unknown column 'projects.status' in 'where clause': SELECT COUNT(*) FROM `dmsf_files` WHERE (projects.status <> 9 AND projects.id IN (SELECT em.project_id FROM enabled_modules em WHERE em.name='dmsf') AND dmsf_files.project_id IN (34) AND dmsf_files.deleted = 0) AND (((LOWER(dmsf_files.name) LIKE '%dmsf%') OR (LOWER(dmsf_file_revisions.title) LIKE '%dmsf%') OR (LOWER(dmsf_file_revisions.description) LIKE '%dmsf%')))):
  lib/redmine/search.rb:127:in `block in load_result_ids'
  lib/redmine/search.rb:125:in `each'
  lib/redmine/search.rb:125:in `load_result_ids'
  lib/redmine/search.rb:115:in `block in load_result_ids_from_cache'
  lib/redmine/search.rb:114:in `load_result_ids_from_cache'
  lib/redmine/search.rb:99:in `result_ids'
  lib/redmine/search.rb:70:in `result_count'
  app/controllers/search_controller.rb:65:in `index'
## end debug-log ##

thx for your help !

**Comment by picman** [comment]
DMSF 1.5.1 is not compatible with Redmine 3.0.3 I'm afraid. Please use devel-1.5.2 instead or just wait for 1.5.2 release (in 1-2 weeks).

**Comment by juocal** [comment]
Hey,

I struggled with the same issue today. And I solved it.

As suggested on the Bitnami FAQ (https://wiki.bitnami.com/Applications/BitNami_Redmine#How_to_install_a_plugin_on_Redmine.3f) I ran: 
`bundle install --without development test postgresql sqlite --no-deployment`

This one failed because of xapian (tar not available for windows), as @u2mike already mentioned.

Thats why I commented out the xapian-full-alavetli part in the gemfile of the plugin with the path:
"installpath\apps\redmine\htdocs\plugins\redmine_dmsf\Gemfile"

After doing that, the build succeeded.

Subsequently I ran:

`ruby bin/rake redmine:plugins RAILS_ENV=production`

... and worked fine! It is running without any error yet and upload is working. Except the indexing of files due to the missing xapian gem. This is already detected by Redmine and thus disabled.

So, see that as solved:

BTW: A workaround to not comment "xapian" out is to run the install with:

`bundle install --without xapian development test postgresql sqlite --no-deployment`
